### DISCUSSION TITLE
Route Integration

### DISCUSSION THREAD

**Discussion by MGatner** [tattersoftware/codeigniter4-assets#30] [Ideas]
Would it be easier to apply Assets directly to the routes via **app/Config/Routes**, similar to how Filters are handled (as Route input)? Or move entirely to Filters with asset groups as input?

```
$routes->add('reports/chart', ['filter' => 'assets:chartjs,jquery,bootstrap']);
```

**Comment by sfadschm** [comment]
In the current state I think it would 👍 
Else we have to write each defined route in the Config/Routes.php and again in Config/Assets.php, if I'm not overlooking something?

**Comment by evansharp** [comment]
I think this is way cleaner too.

**Comment by MGatner** [comment]
Thanks for the feedback all! I have some big changes coming to this library. There will definitely be a Filter and a way of creating asset collections... I haven't ironed out the details about how those interact but it will be a big improvement over the current handlers.

**Comment by MGatner** [comment]
*Note: Issue converted to a Discussion*

Just an update, I've decided to stick with the config file version of `$routes` for now with version 3 mostly due to some limitations with the framework's `Filters`. I am contemplating a Pull Request to the framework to expand some of these features which may result in changes to this library down the road but not for the initial release of version 3.
